server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -144,10 +144,6 @@ ParserContext createMultiFieldContext(ParserContext in) {
                 return new MultiFieldParserContext(in);
             }
 
-            ParserContext createDynamicTemplateFieldContext() {
-                return new DynamicTemplateParserContext(this);
-            }
-
             private static class MultiFieldParserContext extends ParserContext {
```

Comment:
we may want to look into taking the same approach with MultiFieldParseContext, to prevent mistakes in the future.